### PR TITLE
Various bug fixes and tests (main)

### PR DESCRIPTION
**Pull request by alanking**
Addresses #6068
Addresses #6100 
Addresses #6285
Addresses #6365 
Addresses #6428 

Unit tests passed. Core tests are running now.

Confirmed that test for #6428 passes when run in topology from provider and consumer. Noticed that some other tests failed, so I will need to look into these.

### CODE REVIEW

**Comment by alanking** [comment]
Given that some tests are still failing and I'd like to try to make Codacy and cland-tidy as happy as possible, I will revert this to draft so I can make changes freely. Feel free to review anyway.

File: lib/core/include/irods/client_connection.hpp [hunk 1]
```diff
@@ -18,7 +18,9 @@ namespace irods::experimental
     /// authentication occurs.
     ///
     /// \since 4.2.9
-    struct defer_connection {} defer_connection;
+    const inline struct defer_connection
+    {
+    } defer_connection;
```
Comment (korydraughn) [inline lib/core/include/irods/client_connection.hpp:23]: This fix is part of #6673. I meant to pull that out into a separate commit for @epsilon-phase but forgot to do so.

I'll go ahead and do that today (along with any other small changes) and then we can rebase other PRs on top those.

Comment (alanking) [reply lib/core/include/irods/client_connection.hpp:23]: Sounds good

**Comment by alanking** [comment]
Despite how long the output is, there is a single error in clang-tidy. It states that the pointer in the lifetime manager for a `DataObjInfo` is assigned a garbage value in one code path. I am choosing to ignore it because it is predicated on a theoretical code path that isn't even possible (if you read the notes carefully).

If others wish for me to pursue understanding and fixing the complaint, I will, but I don't feel it is worth the time. Otherwise, I will mark this as ready for review once again because the tests seem to be passing now. I am running them again to be sure.

**Comment by korydraughn** [comment]
If you're confident that that code path is impossible and all tests pass, then let's get this merged.

**Review by korydraughn — APPROVED** [review]

**Comment by alanking** [comment]
Just so I feel better/for posterity....
```
 /home/runner/work/irods/irods/server/api/src/rsDataObjRepl.cpp:489:13: note: Assuming the condition is false
        if (source_replica.checksum().empty() && cond_input.contains(VERIFY_CHKSUM_KW)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjRepl.cpp:489:47: note: Left side of '&&' is false
        if (source_replica.checksum().empty() && cond_input.contains(VERIFY_CHKSUM_KW)) {
                                              ^
/home/runner/work/irods/irods/server/api/src/rsDataObjRepl.cpp:489:9: note: Taking false branch
        if (source_replica.checksum().empty() && cond_input.contains(VERIFY_CHKSUM_KW)) {
        ^
/home/runner/work/irods/irods/server/api/src/rsDataObjRepl.cpp:505:13: note: Assuming the condition is false
        if (!source_replica.checksum().empty() && cond_input.contains(REG_CHKSUM_KW)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjRepl.cpp:505:48: note: Left side of '&&' is false
        if (!source_replica.checksum().empty() && cond_input.contains(REG_CHKSUM_KW)) {
                                               ^
```
The code path takes the "false" branch for the condition `source_replica.checksum().empty()` on line 489. The code path then takes the "false" branch for the condition `!source_replica.checksum().empty()` on line 505. It is assuming that `source_replica.checksum().empty()` is false (i.e. `!source_replica.checksum().empty()` == true) and `!source_replica.checksum().empty()` is false, which is a contradiction.

That's not to say that the concern raised here isn't valid, but I don't think we need to worry about it right now given the code path to get there cannot happen as it exists.

Adding #'s

**Comment by korydraughn** [comment]
Very good. That provides evidence that the clang-tidy messages are not valid here.

Proceed with adding the pounds.

**Comment by alanking** [comment]
#'d

**Review by korydraughn — APPROVED** [review]

**Pull request merged**